lib/cache_crispies/json_builder.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module CacheCrispies
+  # Builds out a JSON string directly with Oj::StringWriter
+  class JsonBuilder < HashBuilder
+    # Builds the JSON string with Oj::StringWriter
+    #
+    # @return [Oj::StringWriter]
+    def call(json_writer, flat: false)
+      if @serializer.model.nil?
+        json_writer.push_value(nil)
+        return
+      end
+
+      json_writer.push_object unless flat
+
+      last_nesting = []
+
+      serializer.attributes_by_nesting.each do |nesting, attributes|
+        prefix_length = common_prefix_length(last_nesting, nesting)
+
+        (last_nesting.length - prefix_length).times do
+          json_writer.pop
+        end
+
+        nesting[prefix_length..-1].each do |key|
+          json_writer.push_object(key.to_s)
+        end
+
+        attributes.each do |attrib|
+          write_attribute(json_writer, attrib) if show?(attrib)
+        end
+
+        last_nesting = nesting
+      end
+
+      last_nesting.each do
+        json_writer.pop
+      end
+
+      json_writer.pop unless flat
+
+      json_writer
+    end
+
+    protected
+
+    def write_attribute(json_writer, attribute)
+      # TODO: rescue NoMethodErrors here with something more telling
```

Comment:
Could you explain your reasoning for this comment? What sort of `NoMethodErrors` were you seeing that warranted special handling?

Reply:
I think I copied that in a hurry from https://github.com/codenoble/cache-crispies/blob/main/lib/cache_crispies/hash_builder.rb#L70 🤣 The comments might be worth some clean up.

Reply:
This line of comment seems to have been there since the first version. I guess it is related to reporting of a missing attribute in a model, but never revisited later.